docs/DESIGN.md

```
@@ -1,43 +1,132 @@
 
-To deploy ODH components seamlessly, ODH operator will watch two CRDs:
+To deploy ODH components seamlessly, ODH operator watches two [user-facing custom resources (CRs)](#user-facing-custom-resources-and-their-controllers) - DSCInitialization and DataScienceCluster - which are together responsible for ODH configuration and component setup.
 
-- DSCInitialization
-- DataScienceCluster
+In addition, each ODH component is represented by its own dedicated internal API / custom resource.
+Each component CR singleton instance and its sub-resources are managed by a dedicated component controller.
+More details on the supported ODH components are provided [here](#component-crs-and-controllers).
 
-![Operator Re-architecture](images/Operator%20Architecture.png)
+The operator also watches [Feature Trackers](#featuretrackers) which are cluster-scoped resources in the Data Science Cluster, specifically designed for monitoring and managing objects created via the internal Features API.
```

markdownlint-cli2 (0.17.2)

Link fragments should be valid
null

(MD051, link-fragments)